grader_systems/src/stage/valgrind.rs

```diff
@@ -0,0 +1,1123 @@
+use std::{fs::File, io::Read, path::Path, time::Duration};
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+
+use genos::{
+    output::{Content, Output, RichTextMaker, Section, StatusUpdates, Update},
+    points::PointQuantity,
+    process::{Command, ExitStatus, ProcessExecutor, SignalType, StdinPipe},
+    stage::{StageResult, StageStatus},
+    Executor,
+};
+
+use regex::Regex;
+
+use serde::Deserialize;
+
+use tracing::debug;
+
+// default should be longer than run stage default
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(120);
+
+// exit code used to identify non-signal failures.
+// not configurable via YAML.
+//
+// exit codes >125 are reserved by POSIX; see exit(1p)
+// and https://tldp.org/LDP/abs/html/exitcodes.html
+const ERROR_EXITCODE: i32 = 125;
+
+#[derive(Debug, Default, Deserialize, Clone)]
+pub struct ValgrindConfig {
+    log_file: String,
+    leak_check: Option<bool>,
+    malloc_fill: Option<u8>,
+    free_fill: Option<u8>,
+    suppressions: Option<Vec<String>>,
```

Comment:
Still not sure I see the value in making some of these configurable. We would always want to set leak check to on for this stage right? And the bytes for malloc/free fill are relatively arbitrary. Do you see a use case for allowing the grader to choose which bytes to use for different test cases?

Reply:
Yeah, I agree that making `leak_check`, `malloc_fill`, and `free_fill` non-configurable is a good choice, since leak check should always be on and the bytes are arbitrary. Besides `logfile`, I think `suppressions` might be a good option to keep since it depends on what is allowed. But it would be nice to get rid of all these options and just use "valgrind: true" in the config. Maybe the suppressions can be configurable using one preset file...

Reply:
Yeah, I think it would be worthwhile to just remove all these except supressions, which can be configurable. Not all tests will require a supressions file. 

Also, you're missing points! So you can just have points and an optional supressions list.

Reply:
Right! Definitely should be added. Will update that